packages/smooth_app/lib/pages/product/common/product_list_page.dart

```diff
@@ -318,10 +318,13 @@ class _ProductListPageState extends State<ProductListPage>
         onDismissed: (final DismissDirection direction) async {
           final bool removed = productList.remove(barcode);
           if (removed) {
-            await DaoProductList(localDatabase).put(productList);
             _selectedBarcodes.remove(barcode);
-            setState(() => barcodes.removeAt(index));
+            await DaoProductList(localDatabase).put(productList);
```

Comment:
Why have you moved it?

Reply:
Please it put back the way it was before, which will do exactly the same thing, or state why you changed it.
Have a compassionate thought for the reviewers, that have to ask themselves for each modified line: "why did they do that?".
So if there's no explicit added-value, please rollback that part.